lib/mpl_toolkits/axes_grid1/axes_grid.py

```diff
@@ -32,6 +32,10 @@ def colorbar(self, mappable, *, ticks=None, **kwargs):
         kwargs['userax'] = False
         cb = mpl.colorbar.Colorbar(
             self, mappable, orientation=orientation, ticks=ticks, **kwargs)
+        # If this gets called multiple times on the same axes, some state can
+        # be left on the axes between calls, so for now we will just remove
+        # the callback for all axes_grid colorbars
```

Comment:
This seems like it needs investigation.  You want to add a callback machinery, which I am somewhat leery of because I don't like two-way state. Here it breaks down, so you just remove it, which seems like skirting the problem.    I'm not saying axes_grid1 is doing the right thing, but it seems like a red flag to me that should be understood.